### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by charliethechief**
I did it. yay

### CODE REVIEW

File: js-intro.js [hunk 1]
```diff
@@ -43,15 +59,25 @@ var myCat = "Garfield"
...
-
+var myMessage = "Learning is fun!"
+const myMessage2 = (string) =>{
+let newVar = myMessage
+for(let i=0; i< myMessage.length; i++){
+}
+return newVar
+}
+    console.log(myMessage2(myMessage))
+
```
Comment (sjproctor) [inline js-intro.js:75]: This just returns the string as is. To return every letter you would want to do something like:
const myMessage2 = (string) =>{
  for(let i=0; i< string.length; i++){
    console.log(string[i])
  }
}
myMessage2(myMessage)

File: js-intro.js [hunk 2]
```diff
@@ -43,15 +59,25 @@ var myCat = "Garfield"
...
+return newVar
+}
+    console.log(myMessage2(myMessage))
+    
 // 3b. Write the code that logs each letter of the message using map.
 
-
+var myMessage = "Learning is fun!"
+var newMyMessage = myMessage.split("").map(function(z){return z})
+console.log (myMessage)
```
Comment (sjproctor) [inline js-intro.js:81]: Same thing here. You are logging the string.
var myMessage = "Learning is fun!"
var newMyMessage = myMessage.split("").map(value => value)
console.log(newMyMessage)

File: js-intro.js [hunk 3]
```diff
@@ -69,7 +95,11 @@ var testString = "thisisateststring"
 // 4a. Write a function that takes the variable as an argument and returns the string without vowels. Expected output: "thsstststrng"
 
 
-
+var newString1 = testString.split("")
+const vowler = (x) =>{
+    return x.filter(y =>"aeiou".includes (y.toLowerCase())===false).join("")
+}
+console.log(vowler(newString1))
```
Comment (sjproctor) [inline js-intro.js:103]: Nice!!!

File: js-intro.md [hunk 4]
```diff
@@ -5,49 +5,51 @@ Answer the following questions. First, without external resources. Challenge you
...
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer:the driver is the person activley coding while the controller is watching them code and trying to catch syntax and logical errors
 
-  Researched answer:
+  Researched answer:Pair programming is an agile software development technique in which two programmers work together at one workstation. One, the driver, writes code while the other, the observer or navigator, reviews each line of code as it is typed in. The two programmers switch roles frequently.
```
Comment (sjproctor) [inline js-intro.md:46]: Nice answer!

**Review by sjproctor — APPROVED** [review]
Great job! These are good questions to review as practice. :) Keep it up!

File: js-intro.js (hunk 3 above)
Comment (charliethechief) [reply js-intro.js:103]: Thanks👍🏽

**Pull request merged**